### DISCUSSION TITLE
beta.29 M1 build, install remote kenvs, polish, upcoming lessons

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#353] [Announcements]
I'm starting on lessons/docs on Monday. If you have anything specific you want me to cover, please reply below!

## M1 Build
 If you're on an M1 mac, you can download the new M1 build from https://www.scriptkit.com/

1. Download https://www.scriptkit.com/
2. Quit Kit. *note - typing `kit quit` or `k q` in the app is the fastest way to quit.
3. Drag/drop to overwrite your previous build
4. Kit should now auto-update from the M1 channel
5. Open Kit

## Kenv Management
There are a lot of tools to help manage other kenvs. They're in the `Kit` menu and once you've installed a remote kenv (which is really just a git repo with a scripts dir), then more options show up in the `Edit` menu to move scripts between kenvs, etc. I'll cover this in detail in the docs/lessons

## Polish
Lots of UI work:
* Remembering position - Each script with a `//Shortcut` will remember its last individual prompt position. For example, if you have a script that uses `textarea`, then drag it to the upper right, the next time you launch that script, it will launch in that position.
* `//Image` metadata - Scripts can now have images:
```js
//Image: https://placekitten.com/64
```
or
```js
//Image: logo.png
```
will load from `~/.kenv/assets/logo.png`

* Spinner - added a spinner for when you submit a prompt and the process needs to do some work before opening the next prompt

![CleanShot 2021-07-16 at 12 22 58](https://user-images.githubusercontent.com/36073/125992326-7b6f0034-00e8-41df-9ca7-f0e33becf0b2.gif)

* Resizing - *Lots* of work on getting window resizing behavior consistent between different UIs. This was a huge pain, but you'll probably never appreciate it 😅
* Lots more - many more small things

## Lessons!

I'm starting to work on lessons next week and getting back into streaming schedule. I would ♥️ to hear any specific questions or lessons you would like to see to help you remove some friction from your day. I'll be posting the lessons over on [egghead.io](egghead.io) for your viewing pleasure. Please ask questions in the replies!

**Comment by CoffiDev** [comment]
Hi! I love the tool more with every update, but I have to give myself some time to hackathon some scripts.

Some of the lessons I would want to see are: 
- An overview of all the "meta tags" (the data commented on the top of the script)
- The options to customize the options showed, like adding images, the preview panel
- The known differences on how a scrip behaves on cli vs UI

I sure will enjoy the sessions on egghead!

**Reply by johnlindquist** [reply]
Thanks, great questions. I'll make sure to cover all of that 👍

**Comment by CoffiDev** [comment]
HI there! I just remember another question I had:

What version of node is it using? is it the node I have globally installed on my machine? so for example, I need to have node installed to use skit, or anyone can just download it and start running scripts out of the box?

Also, it could help to have a lesson on creating (and distributing) kenvs, for example, to use it as a tool with your team, jajaja but maybe I'm asking too much already u.u

**Reply by johnlindquist** [reply]
You don't need node installed. Kit.app grabs the latest during the setup process and puts it in ~/.kit/node. 

Kit.app knows to use that version for running scripts, but it's not on your $PATH, so "node --version" will find whatever version you've installed (or none if you haven't).

**Reply by CoffiDev** [reply]
ohhh! great, that way it should be more easy to share automations with other members of your team (not only devs). Thanks for the answer :D
